src/jit/valuenum.cpp

```diff
@@ -1188,7 +1265,7 @@ ValueNum ValueNumStore::VNForMapSelectWork(ValueNumKind vnk, var_types typ, Valu
     VNDefFunc2Arg fstruct(VNF_MapSelect, arg0VN, arg1VN);
     if (GetVNFunc2Map()->Lookup(fstruct, &res))
     {
-        return res;
+        goto Return;
```

Comment:
IMO break and continue are undesirable - they are a lot like early returns. It is too easy to add code at the end of the loop (or method) and not realize that the code will not be executed. Instead, after looking at the code, I believe that you could eliminate the early returns, and in each case where the return is followed by an "if", simply use "else if".